internal/yaml/utils.go

```diff
@@ -0,0 +1,41 @@
+package yaml
+
+import (
+	"fmt"
+	"strings"
+)
+
+const separator = "---"
+
+func ConcatYaml(yaml ...string) string {
+	if len(yaml) == 0 {
+		return ""
+	} else if len(yaml) == 1 {
+		return yaml[0]
+	}
+
+	result := yaml[0]
+	for _, y := range yaml[1:] {
+		result = concatYaml(result, y)
+	}
+	return result
+}
+
+func concatYaml(y1, y2 string) string {
```

Comment:
i wonder if we should eventually store yaml in the data model as a []string, without separators. this is definitely crying out for a more structured representation